### PR TITLE
[DEVHUB-1497] Meta description page no, fix author page dynamic route

### PR DESCRIPTION
**Pull request by ajhenri**
…e setting

## Jira Ticket:

DEVHUB-1497

## Description:

These are fast follow fixes for SEO related findings that Elizabeth shared with me.
They include the following:

- Pull in the same meta description as the main page for the paginated pages and maybe add “topic - page x” in there
- For pages that don't exist, return 404 rather than the first / last page.
- Fix issue with author route which can allow a repeating name in the URL (https://www.mongodb.com/developer/author/sourabh-bagrecha/sourabhbagrecha/)

## Related Issue(s) or Dependencies (if applicable):

Please link to the issue(s) or dependencies here

## Checklist: (Check off where applicable)?

-   [x] I have performed a self-review(QA) of my code and reviewed with Design or Product (If Applicable)?
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have added tests that prove my fix is effective or that my feature works (SEO changes to be tested by Elizabeth)

### CODE REVIEW

**Comment by ajhenri** [comment]
@mcolella14 @justinmckenzie @max-melo This is ready for review if anyone has time. The H1/H2 changes were removed.

File: src/page-templates/topic-content-type-page/topic-content-type-page-template.tsx [hunk 1]
```diff
@@ -96,13 +102,23 @@ export const TopicContentTypePageTemplate: NextPage<
     );
 
     const [pageTitle, setPageTitle] = useState(buildPageTitle(pageNumber));
+    const [metaDescr, setMetaDescr] = useState(
+        description && description !== ''
```
Comment (justinmckenzie) [inline src/page-templates/topic-content-type-page/topic-content-type-page-template.tsx:106]: i could be wrong, but could you get away with something like `useState(description || getMetaDescr(...))`

**Review by justinmckenzie — APPROVED** [review]
only a nitpick that i might be wrong about anyway. LGTM!

**Review by mcolella14 — APPROVED** [review]
Agree with Justin's nitpick, but everything looks great!

**Review by max-melo — COMMENTED** [review]
LGTM!

**Review by max-melo — APPROVED** [review]

**Comment by ajhenri** [comment]
Thanks!

**Pull request merged**